### PR TITLE
build: Improvements to bundled libraries support.

### PR DESCRIPTION
**Pull request by Steveice10**
Ported from the work-in-progress Qt 6 PR, with some added improvements to the Qt installer target.

Introduces aqtinstall for downloading Qt dependencies instead of having to build and distribute binaries ourselves. With this change we can drop both pre-built Qt libraries and pre-built QtIFW for the installer.

Mac builds will still need our own Qt build for now since official Qt 5 builds are not universal. This will be solved with Qt 6.

This also moves the dependency bundling logic for macOS from CI scripts to a CMake build option. This makes it easier to produce working local builds as well as verify functionality without having to run a CI build.

If someone could double check the installer target is working on Windows, would be appreciated. I've already verified on Linux and macOS.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6435)

### CODE REVIEW

**Comment by Steveice10** [comment]
Rebased and ported over the rest of the build system changes not specific to Qt 6 from my other PR, so that we can have this PR focused on the build system improvements and the Qt 6 PR focused on the actual Qt update.

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,7 +38,7 @@ if (ENABLE_FFMPEG_AUDIO_DECODER OR ENABLE_FFMPEG_VIDEO_DUMPER)
     set(ENABLE_FFMPEG ON)
 endif()
 
-CMAKE_DEPENDENT_OPTION(CITRA_USE_BUNDLED_FFMPEG "Download bundled FFmpeg binaries" ON "ENABLE_FFMPEG;MSVC" OFF)
+CMAKE_DEPENDENT_OPTION(CITRA_USE_BUNDLED_FFMPEG "Download bundled FFmpeg binaries" ON "ENABLE_FFMPEG;MSVC OR APPLE" OFF)
```
Comment (SachinVin) [inline CMakeLists.txt:41]: Wouldn't it be preferable for macos users to get ffmpeg(also qt) from homebrew instead?

Comment (Steveice10) [reply CMakeLists.txt:41]: Homebrew does not (and will not) vend universal libraries which makes cross compiling and making distributable builds more difficult. Plus this makes local build behave more consistent with CI builds.

Comment (SachinVin) [reply CMakeLists.txt:41]: Okay

**Review by GPUCode — APPROVED** [review]

**Comment by Steveice10** [comment]
Rebased

**Pull request merged**